### PR TITLE
add list_keys() keygrip attribute

### LINKED ISSUE #166: list_keys() show keygrip
**Is your feature request related to a problem? Please describe.**
I want to find out what keygrip a key has. 

**Describe the solution you'd like**
The `--list-keys` and `--list-secret-keys` command can be called with the `--with-keygrip` option.

### PR DESCRIPTION
**Pull request by kilimnik**
I implemented the already discussed feature request in #166.

I know it isn't backwards compatible. If that's a problem, I can try to fix that. I am a bit unsure how because not setting the `--with-keygrip` arg is not enough, because the keygrip is also provided when using `--list-secret-keys` in combination with `--with-colons`. It would be necessary to remove the `grp` arg from the `valid_keywords`.

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -476,53 +476,55 @@ compatibility reasons. As `GnuPG documentation
...
++-------------+--------------------------------------------------------------------------------+
+| length      | The length of the key in bits                                                  |
++-------------+--------------------------------------------------------------------------------+
+| algo        | Public key algorithm                                                           |
++-------------+--------------------------------------------------------------------------------+
+| keyid       | The key ID                                                                     |
++-------------+--------------------------------------------------------------------------------+
+| date        | The creation date of the key in UTC as a Unix timestamp                        |
++-------------+--------------------------------------------------------------------------------+
+| expires     |                                                                                |
```
Comment (vsajip) [inline docs/index.rst]: Why is the value column for this row empty?

File: docs/index.rst [hunk 2]
```diff
@@ -598,6 +600,10 @@ user-friendly, but nevertheless it should be usable.)
    suitable for converting to an ``error_map`` value (basically, converting the string
    keys in the JSON to integers using base 16).
 
+.. versionadded:: 0.4.9
+   Information on keys returned by :meth:`list_keys` now includes the ``keygrip`` attribute.
+   The ``subkeys`` attribute now also consist of 4 values with the ``keygrip`` being the 4th.
+
```
Comment (vsajip) [inline docs/index.rst:607]: Please mention GnuPG version compatibility information here.

File: gnupg.py [hunk 3]
```diff
@@ -1428,7 +1435,7 @@ def list_keys(self, secret=False, keys=None, sigs=False):
             which = 'secret-keys'
         else:
             which = 'sigs' if sigs else 'keys'
-        args = ['--list-%s' % which, '--fingerprint', '--fingerprint']  # get subkey FPs, too
+        args = ['--list-%s' % which, '--with-keygrip', '--fingerprint', '--fingerprint']  # get subkey FPs, too
```
Comment (vsajip) [inline gnupg.py]: Please make this more compatible by e.g. inserting the `--with-keygrip` only for GnuPG >= 2.1. I think the `FIELDS` and `valid_keywords` may not need to be conditional on GnuPG version - not sure.

File: test_gnupg.py [hunk 4]
```diff
@@ -545,13 +546,14 @@ def test_list_subkey_after_generation(self):
...
         self.assertEqual(key_info['subkeys'][1][2], subkey_encrypt.fingerprint)
-        for skid, _, sfp in key_info['subkeys']:
+        for skid, _, sfp, grp in key_info['subkeys']:
             self.assertTrue(skid in skinfo)
             info = skinfo[skid]
             self.assertEqual(info['keyid'], skid)
             self.assertEqual(info['type'], 'sub')
             self.assertTrue(sfp in public_keys.key_map)
             self.assertTrue(public_keys.key_map[sfp] is key_info)
+            self.assertIsNotNone(grp)
```
Comment (vsajip) [inline test_gnupg.py]: Could this check be stronger? For example, `self.assertTrue(grp)` would check for a false value in general, such as an empty string. Same in other places where `assertIsNotNone` is used.

**Review by vsajip — CHANGES_REQUESTED** [review]
Please retain backward compatibility with GnuPG 1.x.

File: docs/index.rst (hunk 1 above)
Comment (vsajip) [reply docs/index.rst]: Wait, I think this is a GitHub rendering bug.

Comment (kilimnik) [reply docs/index.rst]: No you're right, not sure what happened here

**Pull request merged**